File: ProjetPI(back)/src/main/java/tn/esprit/backend/entite/message.java
package tn.esprit.backend.entite;

import jakarta.persistence.*;
import lombok.*;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
public class message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long idMessage;

    @ManyToOne
    user user;

    private String content;

    @Temporal(TemporalType.TIMESTAMP)
    private Date sentAt;

}

File: ProjetPI(front)/src/app/Models/message.ts
import {User} from "./User";

export interface Message {
  idMessage: number;
  sender: User;
  receiver: User;
  content: string;
  date: Date;
  reactions: string[];
  showDateTime: boolean;
  file: File;
  isFile: boolean;
  fileName: string;
}
